src/plugins/data/public/ui/dataset_selector/dataset_table.tsx

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiBasicTable, EuiFieldSearch, EuiLink, EuiText } from '@elastic/eui';
+import React, { useRef, useState } from 'react';
+import { FormattedMessage } from '@osd/i18n/react';
+import { DataStructure } from '../../../common';
+import { getQueryService } from '../../services';
+import { DataStructureFetchOptions } from '../../query';
+
+interface DatasetTableProps {
+  path: DataStructure[];
+  setPath: (newPath: DataStructure[]) => void;
+  index: number;
+  explorerDataset: DataStructure | undefined;
+  selectDataStructure: (item: DataStructure | undefined, newPath: DataStructure[]) => Promise<void>;
+  fetchNextDataStructure: (
+    nextPath: DataStructure[],
+    dataType: string,
+    options?: DataStructureFetchOptions
+  ) => Promise<DataStructure>;
+}
+
+export const DatasetTable: React.FC<DatasetTableProps> = (props) => {
+  const datasetService = getQueryService().queryString.getDatasetService();
+  const [loading, setLoading] = useState(false);
+  const searchRef = useRef<HTMLInputElement | null>(null);
+
+  const initialSelectedIds = props.explorerDataset?.id.split(',');
+  const dataStructures = props.path[props.index].children || [];
+  const paginationToken = props.path[props.index].paginationToken;
+
+  const onTableChange = async (options: DataStructureFetchOptions) => {
+    const typeConfig = datasetService.getType(props.path[1].id);
+    if (!typeConfig) return;
+
+    setLoading(true);
+    await props
+      .fetchNextDataStructure(props.path, typeConfig.id, options)
```

Comment:
not a review but a question. taking in consideration potential amount of data structures that could return and pages. do you think we need to have some limit on this?

like an advanced setting that lets users set max page size default to a value like 10 and then max pages? that way we prevent the ability to just manually scan through 100s of pages for a tables and force the user to use the search bar.

not positive if the data source that is being queried could incur a lot of costs.

Reply:
the OUI table component is a bit weird that the selections get cleared if items gets unloaded (i.e. when user go to the next page, search by something else, etc). 

So the issue with search alone is that if user wants to select `foo`, `bar`, after searching and selecting `foo`, searching `bar` would clear the selection. i haven't figured out a good UX for this